org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/markers/ApplyQuickFixMarkerResolver.java

```diff
@@ -65,31 +72,58 @@ public void run(IMarker marker) {
     if (file == null) {
       return;
     }
-    IDocument document = getDocument(file);
+    ITextEditor openEditor = openEditor(file, marker);
     Display.getDefault().asyncExec(() -> {
-      applyIn(document, fix);
-      SonarLintCorePlugin.getTelemetry().addQuickFixAppliedForRule(MarkerUtils.getRuleKey(marker).toString());
+      boolean previousValue = disableAutoAnalysis(file.getProject());
+      try {
+        applyIn(openEditor, fix);
+        SonarLintCorePlugin.getTelemetry().addQuickFixAppliedForRule(MarkerUtils.getRuleKey(marker).toString());
+      } finally {
+        restoreAutoEnabledState(file.getProject(), previousValue);
+      }
     });
   }
 
-  private static IDocument getDocument(ISonarLintFile file) {
-    IDocument doc;
-    IEditorPart editorPart = PlatformUtils.findEditor(file);
-    if (editorPart instanceof ITextEditor) {
-      doc = ((ITextEditor) editorPart).getDocumentProvider().getDocument(editorPart.getEditorInput());
+  private static boolean disableAutoAnalysis(ISonarLintProject project) {
+    SonarLintProjectConfiguration projectConfig = SonarLintCorePlugin.loadConfig(project);
+    boolean previousValue = projectConfig.isAutoEnabled();
+    projectConfig.setAutoEnabled(false);
+    SonarLintCorePlugin.saveConfig(project, projectConfig);
+    return previousValue;
+  }
+
+  private static void restoreAutoEnabledState(ISonarLintProject project, boolean previousValue) {
+    SonarLintProjectConfiguration projectConfig2 = SonarLintCorePlugin.loadConfig(project);
+    projectConfig2.setAutoEnabled(previousValue);
+    SonarLintCorePlugin.saveConfig(project, projectConfig2);
```

Comment:
I was a bit surprised that we modify real user configuration to achieve this, it seems strange to play with user-facing values to make sure parallel analyses don't happen, I think it's more an internal concern. Why do we need to do this ?